### DISCUSSION TITLE
The BCs of totalPressure and totalTemperature can be used in DARhosimpleCFoam?

### DISCUSSION THREAD

**Discussion by NUAA-qinfei** [mdolab/dafoam#675] [General]
I know in the latested DAFoam v3.1.0 allowed using totalPressure and totalTemperature as the boundary conditions for turbomachinery cases，so I want use this BCs for a turbofan engine ,the engine inlet I use static pressure BC, the hot-outlet and the cold-outlet I use  totalPressure and totalTemperature as BCs, the far-field Ma=0.8, so I use  DARhosimpleCFoam  because of the  transonic flow, and I use S-A 
model,but it has erroes:"Invalid values found. Return primal failure and reset the states to their initial values ", however,when I use the DARhosimpleFoam, it works normally,so I have a question:The BCs of totalPressure and totalTemperature can be used in DARhosimpleCFoam? or can I use the  DARhosimpleFoam for transonic flow?
![微信图片_20240830213440](https://github.com/user-attachments/assets/2453b522-0868-416e-912e-53ab62202bd0)
![微信图片_20240830214158](https://github.com/user-attachments/assets/f8dc0a2c-6518-446e-82ec-ba9aefda28f6)

**Comment by friedenhe** [comment]
The error you saw was caused by flow divergence. No, you are supposed to use DARhoSimpleCFoam for transonic conditions. Otherwise, the flow may either diverge or return unphysical results. Regarding your setup, you need to provide more information. How did you model the engine? Are you using an actuator disk model to simulate the engine?

**Reply by NUAA-qinfei** [reply]
Thanks for Prof's reply,I just use 0-dimensional simulation to model the turbofan engine,not same as the pycycle, so I don't use the actuator disk model, I just use the intake and exhaust BCs to model the engine thrust ,the setup can be shown in this picture,this setup converge very well in the business software Ansys Fluent or my own CFD code. I want to know if DAFoam support this intake and exhaust simulation?I have not see anyone do this in DAFoam, previously,I use the DAFoam to do aerodynamic adjoint optimize for a wing or aircraft,it works very well,now I just want to consider the intake and exhaust effect for the aerodynamic adjoint optimize.
On the other hand, this flow divergence maybe caused by poor Mesh quality?because I have some nonOrthoFaces in the DAFoam mesh check,I know the OpenFoam is sensitive to mesh quality,especially for the unstructured mesh.
Thanks to Prof Ping He and DAFoam,I benefit a lot from the DAFoam discussion.
![微信图片_20240904104133_副本](https://github.com/user-attachments/assets/ed3e20ea-2eee-4e71-8ad0-127bd24fe649)
![Uploading Screenshot from 2024-09-04 10-05-17.png…]()
[BrunScript.txt](https://github.com/user-attachments/files/16860166/BrunScript.txt)
[0.zip](https://github.com/user-attachments/files/16860167/0.zip)

**Reply by NUAA-qinfei** [reply]
![Screenshot from 2024-09-04 10-05-17](https://github.com/user-attachments/assets/39746a42-66b6-436d-87bb-44f70f076f56)

**Comment by friedenhe** [comment]
Your BCs for the engine look good to me. I am not 100% sure what caused the flow divergence. But I suggest you separate the engine and wing (i.e., simulate only the wing and simulate only the engine) and see if the separated simulation converges. Another thing you can try is to use a non-uniform flow field as the initial condition, which should help the convergence.

**Comment by NeuroNexusAI** [comment]
"Absolutely! Using the BCs for total Pressure and total Temperature in DARho Simple C Foam is key for accurate simulations. They enhance flow accuracy and performance. Let’s optimize those results!"
